File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    answer = []
    
    for command in commands:
        new_array = array[command[0]-1: command[1]]  # 슬라이싱으로 배열을 자름
        new_array.sort()  # 정렬, 반환값이 None이 아님
        answer.append(new_array[command[2] - 1]) # 정렬된 리스트 출력
    
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        for (int idx = 0; idx < commands.length; idx++) {
            int i = commands[idx][0] - 1; // 배열은 0-based index
            int j = commands[idx][1];
            int k = commands[idx][2] - 1;

            // 배열 자르기
            int[] subArray = Arrays.copyOfRange(array, i, j);

            // 정렬
            Arrays.sort(subArray);

            // k번째 수 저장
            answer[idx] = subArray[k];
        }

        return answer;
    }
}
